src/uct/ib/mlx5/exp/ib_exp_md.c

```diff
@@ -492,9 +495,9 @@ static ucs_status_t uct_ib_mlx5_exp_reg_multithreaded(uct_ib_md_t *ibmd,
     struct ibv_mr *umr;
     int mr_num;
 
-    if (!(md->super.dev.dev_attr.comp_mask & IBV_EXP_DEVICE_ATTR_COMP_MASK_2) ||
-        !(md->super.dev.dev_attr.comp_mask_2 & IBV_EXP_DEVICE_ATTR_UMR_FIXED_SIZE_CAPS) ||
-        !(md->super.dev.dev_attr.exp_device_cap_flags & IBV_EXP_DEVICE_UMR_FIXED_SIZE)) {
+    if (!(md->super.dev.dev_attr.comp_mask & (uint32_t)IBV_EXP_DEVICE_ATTR_COMP_MASK_2) ||
```

Comment:
i think it should be uint64_t
BTW can we unit common code with uct_ib_mlx5_exp_reg_atomic_key: uct_ib_mlx5_md_is_ksm_supported() for the if below

Reply:
fixed + unified into common function